src/apps/recentlyUsed.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Gets the list of apps that have been used recently by the authenticated user.
+ *
+ * @module apps/recentlyUsed
+ */
+
+import { validateInterval, validateLimit } from "../util";
+import * as config from "../configuration";
+import constants from "../constants";
+import logger from "../logging";
+
+// All apps returned by this query are DE apps, so the system ID can be constant.
+const recentlyUsedAppsQuery = `
+    SELECT DISTINCT
+        a.id,
+        'de' AS system_id,
+        a.name,
+        a.description,
+        a.wiki_url,
+        a.integration_date,
+        a.edited_date,
+        u.username,
+        EXISTS (
+            SELECT * FROM users authenticated_user
+            JOIN workspace w ON authenticated_user.id = w.user_id
+            JOIN app_category_group acg ON w.root_category_id = acg.parent_category_id
+            JOIN app_category_app aca ON acg.child_category_id = aca.app_category_id
+            WHERE authenticated_user.username = $1
+            AND acg.child_index = $2
+            AND aca.app_id = a.id
+        ) AS is_favorite,
+        TRUE AS is_public,
```

Comment:
Is this still correct? Private apps should appear here now, I think, since public apps aren't the scope.

Reply:
Ah, thanks for catching that. It's not true any longer.

Reply:
This should be fixed now. I also did a little refactoring so that we only get the list of public app IDs once for every request to `/users/:username`.